Implement telemetry reporting (#6619)
RasaHQ/rasa: docs/docs/telemetry.mdx

---
id: telemetry
sidebar_label: Rasa Telemetry
title: Rasa Telemetry
---

Rasa uses telemetry to report anonymous usage information. This information is

> essential help improve Rasa Open Source for all users.

Reviewer: missing 'to': essential to help improve

For the team working on Rasa Open Source it is important to understand how

> the product is used, it allows us to properly prioritize our research efforts

Reviewer: Just a grammatical thing: comma splice. Should be 2 sentences:

For the team working on Rasa Open Source it is important to understand how
the product is used. It allows us to properly prioritize our research efforts

and feature development.

You will be notified about the telemetry reporting when running Rasa Open Source
for the first time.

## How to opt-out

> You can always opt-out from telemetry reporting using

Reviewer: You can always opt-out => You can opt out of telemetry reporting at any time using...

If you want to enable reporting again, you can use
```bash
rasa telemetry enable
```

## Why do we use telemetry reporting?

**Anonymous** telemetry data allow us to prioritize our research efforts
and feature development based on usage. We want to collect aggregated
information on usage and reliability so that we can ensure a high-quality product.

So how will we use the reported telemetry data? Here are some examples
what we use the data for:

- We will be able to know which languages, pipelines and policies are used.
  This will enable us direct our research efforts towards projects
  that have the most impact on text and dialogue handling.
- We will be able to know data set sizes and general structure (e.g. the number
  of intents). This allows us to better test our software on different types
  of data sets and optimize the frameworks performance.
- We will be able to get more detail on the types of errors you are running
  into while building an assistant (e.g. initialization, training, etc.).
  This will let us improve the quality of our framework and better focus our
  time on solving more common, frustrating issues.

> 

Reviewer: I'd suggest moving the sensitive data section up higher, after the ## Why do we use telemetry reporting? section. 

Your first thought after you read the examples of data usage is whether Rasa can read your training data and user messages. Maybe even renaming the section ## What do we *not* report?

---
id: telemetry
sidebar_label: Rasa Telemetry
title: Rasa Telemetry
---

> Rasa uses telemetry to report anonymous usage information. This information is

Reviewer 2: can you put this paragraph or at least part of it under `abstract:` below the title above? If you try doing `make livedocs` you'll see how that looks

Author: done :+1:

essential to help improve Rasa Open Source for all users.
For the team working on Rasa Open Source it is important to understand how
the product is used. It allows us to properly prioritize our research efforts
and feature development.

You will be notified about the telemetry reporting when running Rasa Open Source
for the first time.

## How to opt-out

> You can opt out of telemetry reporting at any time using

Reviewer 2: ```suggestion
You can opt out of telemetry reporting at any time by running the command:
```

```bash
rasa telemetry disable
```

or by setting `RASA_TELEMETRY_ENABLED=false` as an environment variable.

> If you want to enable reporting again, you can use

Reviewer 2: ```suggestion
If you want to enable reporting again, you can run:
```

```bash
rasa telemetry enable
```

## Why do we use telemetry reporting?

**Anonymous** telemetry data allow us to prioritize our research efforts
and feature development based on usage. We want to collect aggregated
information on usage and reliability so that we can ensure a high-quality product.

So how will we use the reported telemetry data? Here are some examples
what we use the data for:

- We will be able to know which languages, pipelines and policies are used.
  This will enable us direct our research efforts towards projects

>   that have the most impact on text and dialogue handling.

Reviewer 2: i'm not sure this sentence makes sense - do we mean something like "This will enable us to direct our research efforts towards text and dialogue handling projects that will have the biggest impact for our users"?

- We will be able to know data set sizes and general structure (e.g. the number
  of intents). This allows us to better test our software on different types
  of data sets and optimize the frameworks performance.
- We will be able to get more detail on the types of errors you are running
  into while building an assistant (e.g. initialization, training, etc.).
  This will let us improve the quality of our framework and better focus our
  time on solving more common, frustrating issues.

## What about sensitive data?

> Your sensitive data never leaves your machine. We

Reviewer 2: ```suggestion
Your sensitive data never leaves your machine. We:
```

- don't report any personal identifiable information
- don't report your training data

> - don't report any messages your assistant receives or sends

Reviewer 2: can we maybe bold the `don't` in all of these?

:::note Inspect what is reported
You can view all the telemetry information that is reported

> by setting the environment variable `RASA_TELEMETRY_DEBUG=true`, e.g.

Reviewer 2: ```suggestion
by defining the environment variable `RASA_TELEMETRY_DEBUG=true`, for example when running the train command:
```

```bash
RASA_TELEMETRY_DEBUG=true rasa train
```

> When you set `RASA_TELEMETRY_DEBUG` no information will be send to any server,

Reviewer 2: ```suggestion
When you set `RASA_TELEMETRY_DEBUG` no information will be sent to any server,
```

> instead it will be logged to the commandline as a json dump for you to inspect.

Reviewer 2: ```suggestion
instead it will be logged to the commandline as a json dump for you to inspect.
:::
```
you need to finish a note with `:::` now

## What do we report?

Rasa reports aggregated usage details, command invocations, performance
measurements and errors.
We use the telemetry data to better understand usage patterns. The reported data
will directly allow us to better decide how to design future features
and prioritize current work.

Specifically, we collect the following information for all telemetry events:

> - Type of the reported event (e.g. `Training Started`)

Reviewer 2: this isn't code, so it shouldn't be formatted with ` - maybe italics instead?

> - Rasa machine ID: This is generated with UUID and stored in the global Rasa

Reviewer 2: ```suggestion
- Rasa machine ID: This is generated with a UUID and stored in the global Rasa
```

  config at `~/.config/rasa/global.yml` and sent as `metrics_id`

> - One-way hash of the current working directory or a hash of the git remote (`project`)

Reviewer 2: what does (`project`) mean?

- General OS level information (operating system, number of CPUs, number of
  GPUs and whether the command is run inside a CI)
- Current Rasa Open Source and Python version

Here is an example report that shows the data reported to Rasa after running
`rasa train`:
```json
{
  "userId": "38d23c36c9be443281196080fcdd707d",
  "event": "Training Started",
  "properties": {
    "language": "en",
    "num_intent_examples": 68,
    "num_entity_examples": 0,
    "num_actions": 17,
    "num_templates": 6,
    "num_slots": 0,
    "num_forms": 0,
    "num_intents": 6,
    "num_entities": 0,
    "num_story_steps": 5,
    "num_lookup_tables": 0,
    "num_synonyms": 0,
    "num_regexes": 0,
    "metrics_id": "38d23c36c9be443281196080fcdd707d"
  },
  "context": {
    "os": {
      "name": "Darwin",
      "version": "19.4.0"
    },
    "ci": false,
    "project": "a0a7178e6e5f9e6484c5cfa3ea4497ffc0c96d0ad3f3ad8e9399a1edd88e3cf4",
    "python": "3.7.5",
    "rasa_open_source": "2.0.0",
    "gpu": 0,
    "cpu": 16
  }
}
```

> We **cannot identify individual users** from the dataset. It is anonymized and

Reviewer 2: can you add "anonymized" to the typo CI exclusion list?
